back/app/services/insights/voting_phase_insights_service.rb

```diff
@@ -1,7 +1,113 @@
 module Insights
   class VotingPhaseInsightsService < IdeationPhaseInsightsService
+    def vote_counts_with_user_custom_field_grouping(custom_field = nil)
+      ideas = @phase.ideas.transitive # TODO: This may not be precise enough
+      participations = cached_phase_participations
+      voting_participations = participations[:voting]
+      offline_votes = @phase.manual_votes_count
+      online_votes = if @phase.voting_method == 'budgeting'
+        voting_participations.sum { |p| p[:ideas_count] }
+      else
+        voting_participations.sum { |p| p[:total_votes] }
+      end
+
+      total_votes = online_votes + offline_votes
+
+      options = if custom_field&.options&.any?
+        custom_field.options.map do |opt|
+          { "#{opt.key}": { id: opt.id, title_multiloc: opt.title_multiloc }, ordering: opt.ordering }
+        end
+      else
+        []
+      end
+
+      {
+        online_votes: online_votes,
+        offline_votes: offline_votes,
```

Comment:
We seem to be switching language between `total_online_votes` and `online_votes`